raiden/tests/utils/transfer.py

```diff
@@ -268,6 +269,37 @@ def assert_synched_channel_state(
     assert_mirror(channel1, channel0)
 
 
+def wait_synched_channel_state(
+        token_network_identifier,
+        app0,
+        balance0,
+        pending_locks0,
+        app1,
+        balance1,
+        pending_locks1,
+        timeout=5,
+):
+    """Like assert_synched_channel_state, but retry and return True or False instead of throwing"""
+    start_time = time.time()
+    while True:
+        try:
+            assert_synched_channel_state(
+                token_network_identifier,
+                app0,
+                balance0,
+                pending_locks0,
+                app1,
+                balance1,
+                pending_locks1,
+            )
+        except AssertionError:
+            if timeout and time.time() > start_time + timeout:
```

Comment:
why not use `gevent.wait()` here?

Reply:
Isn't `gevent.wait` intended to be used with list of greenlets? This method is just a tiny wrapper around the above `assert_` method, to give time for the state channels to be synched, and preserve the stack trace in the case it keeps failing. As the `assert_synched_channel_state` is synchronous and raises `AssertionError` in case something is not ready, this loop will use it to avoid duplicating the whole code with `if`s.

Reply:
(facepalm.jpg) I meant `gevent.timeout` of course.

Reply:
I don't want to wait for an operation, it always is instantanious. I want to retry while it's failing, until it succeeds, and only then throw the last traceback, which tells a lot what happened. I may be missing something, could you give an example of how it could be done with `gevent.Timeout`? Anyway, did a small refactor to make it more generic and support any `assert_*` function